### PR TITLE
Performance changes to OpenGL render block

### PR DESCRIPTION
**Pull request by JoeMatt**
### What does this PR do

Replaces #1177 with recommendations from @braindx 
Performance changes to OpenGL render block

### Where should the reviewer start

Test graphics rendering for all cores. Non-gl, native open gl and double-buffered cores.

### What are the relevant tickets

#1177

### CODE REVIEW

**Comment by braindx** [comment]
Would probably be good to change hasDepth in the renderBlock to coreRendersToOpenGL or something like that. The emulator core itself using OpenGL is really what requires those additional calls since you can no longer rely on the GL state being how the Provenance last left it.

**Comment by JoeMatt** [comment]
> Would probably be good to change hasDepth in the renderBlock to coreRendersToOpenGL or something like that. The emulator core itself using OpenGL is really what requires those additional calls since you can no longer rely on the GL state being how Provenance last left it.

Cool, I can make that tweak. By the way, I just learned on GitHub you can review a PR and the first icon on a line comment is a code suggestion box, you can select single or multiple lines and suggest the code change directly and I can just click a button to merge them.

Appreciate the feecback!

File: Provenance/Emulator/PVGLViewController.m [hunk 1]
```diff
@@ -487,13 +487,15 @@ - (void)glkView:(GLKView *)view drawInRect:(CGRect)rect
 
     MAKEWEAK(self);
 
-    void (^renderBlock)(void) = ^()
+    void (^renderBlock)(BOOL hasDepth) = ^()
```
Comment (braindx) [inline Provenance/Emulator/PVGLViewController.m]:
```suggestion
    void (^renderBlock)(BOOL doesCoreRenderToOpenGL) = ^()
```

File: Provenance/Emulator/PVGLViewController.m [hunk 2]
```diff
@@ -529,9 +531,11 @@ - (void)glkView:(GLKView *)view drawInRect:(CGRect)rect
             glUniform1i(strongself->blitUniform_EmulatedImage, 0);
         }
         
-        glDisable(GL_DEPTH_TEST);
-        glDisable(GL_CULL_FACE);
-        
+        if (!hasDepth) {
```
Comment (braindx) [inline Provenance/Emulator/PVGLViewController.m]:
```suggestion
        if (!doesCoreRenderToOpenGL) {
```

**Comment by braindx** [comment]
Very cool. I didn't know about that, thanks! I just did it to check out how it works.

**Comment by JoeMatt** [comment]
Nice, merge worked, now if only CI worked to error test, lol. At least i can pull and test from command line easily.

**Comment by mrjschulte** [comment]
Did this ever get merged in?

**Comment by mrjschulte** [comment]
Oh btw, all of the new merges above resulted in a lower fps than what we had before on older ArmV7 iPad hardware. :(

**Comment by braindx** [comment]
Which merges?

**Comment by mrjschulte** [comment]
@braindx, my original (albeit rough) proposal in #1177 was to speed up the GL render block for smoother playback performance on lower spec devices. 

The current state of the Pv develop branch, with merges of the resolution to that original PR with this one that Joe created, has sadly resulted in a performance degradation somewhat defeating the intent and purpose of the original PR :(

**Comment by braindx** [comment]
The changes made to this PR since submission should have identical performance when running a non-OpenGL core in Release compared to your original PR. At worst, when running an OpenGL core in Debug it will have identical performance as before (without the PR).

Any degradation in performance would be due to commits in the develop branch between when this PR was submitted and now. If that's the case, singling out which commits caused a performance regression would be helpful. I'm not familiar with what has been committed recently.

**Comment by mrjschulte** [comment]
@JoeMatt did this get merged back into the develop branch yet? Doesn't look like it by the latest pull. 

@braindx if this hasn't been pulled in yet then regard my other comments.

**Comment by mrjschulte** [comment]
I will look at rebasing this PR against our current develop and refactor it based on the outcome of this conversation. Thanks guys!

**Comment by mrjschulte** [comment]
Suggestions implemented partially in #1454 Closing this Pr.

**Pull request closed without merging**